web/src/main/java/com/wilma/web/controller/web/portal/PartnerPortalController.java

```diff
@@ -28,26 +46,146 @@ public String dashboard(Model model) {
+                "reply", replyDTO));
+
+        log.info("Reply added: "+ reply);
+        return new RedirectView("/partner/forum-thread?category="+ category);
+    }
+
+    @PostMapping("/create-post")
+    public RedirectView createPost(@ModelAttribute PostDTO postDTO, Model model){
+        var newPost = forumService.addPostFromDTO(postDTO);
+        var categoryName = newPost.getCategory().getName();
+        model.addAllAttributes(Map.of(
+                "currentPage", "forum",
+                "menuElements", UserConfiguration.partnerMenuElements,
+                "availableCategories", categoryService.findAll(),
+                "availableTags", tagService.findAll(),
+                "categoryName", categoryName,
+                "postsByCategory", forumService.getPostByCategoryName(categoryName),
+                "repliesForPosts", forumService.getPostRepliesByCategory(categoryName),
+                "post", postDTO));
+
+        log.info("Post created from DTO: "+ newPost);
+        return new RedirectView("/partner/forum");
+    }
+
+    @GetMapping("/forum-thread")
+    public String forumThread(@RequestParam String category, Model model) {
+        model.addAllAttributes(Map.of(
+                "currentPage", "forum",
+                "menuElements", UserConfiguration.partnerMenuElements,
+                "categoryName", category,
+                "postsByCategory", forumService.getPostByCategoryName(category),
+                "repliesForPosts", forumService.getPostRepliesByCategory(category)));
+        return "/partner/forum/forum-thread";
+
+    }
+    //endregion
+
+    //region Expressions of interest
+    @GetMapping("/expressions-of-interest")
+    public String expressionsOfInterest(Model model) {
+        model.addAllAttributes(Map.of(
+                "currentPage", "Expressions Of Interest",
+                "menuElements", UserConfiguration.partnerMenuElements,
+
+                "pendingPartnerExpressionsOfInterest", List.of(
+                        new ExpressionOfInterest(1, new Partner("Microsoft", "Microsoft"), new Date(), new Date(), Period.of(0,0,1), "Brisbane", "A sample job", false, false),
+                        new ExpressionOfInterest(2, new Partner("Google", "Google"), new Date(), new Date(), Period.of(0,11,1), "Perth", "Another sample job", false, false),
+                        new ExpressionOfInterest(3, new Partner("Apple", "Apple"), new Date(), new Date(), Period.of(1,0,0), "Sydney", "A placement example", false, false),
+                        new ExpressionOfInterest(4, new Partner("Amazon", "Amazon"), new Date(), new Date(), Period.of(1,1,1), "Melbourne", "Slavery with extra steps", false, false)
+                )
+        ));
+        return "/partner/expressions-of-interest";
+    }
+    //endregion
+
+    //region Profile
+    @GetMapping("/profile")
+    public String partnerProfile(Model model) {
+        model.addAllAttributes(Map.of(
+                "currentPage", "Profile",
```

Comment:
Menu item not being selected
```suggestion
"currentPage", "profile",
```